src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RuntimeMetrics.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+#if NET
+using System.Threading;
+#endif
+
+namespace System.Diagnostics.Metrics
+{
+    internal static class RuntimeMetrics
```

Comment:
I'd suggest we make this metrics only supported in 9.0+ to avoid the dependency issues targetting downlevel versions. I think a decent way to do that would be adding something like this in the PropertyGroup within the .csproj:
```
<DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net9.0'))">$(DefineConstants);FEATURE_RUNTIME_METRICS</DefineConstants>
```

And then making all the new code conditional on the FEATURE_RUNTIME_METRICS define. 

fyi @tarekgh in case there is a more preferred way of doing this.

Reply:
if ifdef only few lines of the code use `#if NET9_0_OR_GREATER`. If you want to ifdef the whole file, it is preferred to use something like:

```Xml
  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net9.0'))">
    <Compile Include="System\Diagnostics\Metrics\RuntimeMetrics.cs" />
  </ItemGroup>
```

Reply:
> @noahfalk are you suggesting moving all new runtime metrics to 9.0?

I'm suggesting the entire Meter be 9.0 only. Otherwise its not just the cpu.time metric that will need special-casing. A variety of the other instruments have APIs will have varying support depending on FX vs .NET 6 vs. .NET 8. I didn't want to with varying platform support on an instrument-by-instrument basis.

Reply:
This means users will continue to use OpenTelemetry instrumentation library on down-levels for the runtime metrics?

Reply:
yep! (or a few other options depending on their goals)

Reply:
I've updated the PR to limit this to .NET 9+, as suggested. I wonder if we could support .NET 8+ instead since that's LTS? That way, consumers could grab the latest package to have almost all metrics aligned with the semantic conventions. All APIs the implementation depends on are available for .NET 8, except for the new `Environment.CpuUsage`, which could be conditionally excluded as a .NET 9+ only metric.